py/desiutil/annotate.py

```diff
@@ -0,0 +1,376 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+"""
+=================
+desiutil.annotate
+=================
+
+Tools for adding units and comments to FITS files.
+"""
+import csv
+import os
+import sys
+from argparse import ArgumentParser
+import yaml
+from astropy.io import fits
+from astropy.table import Table, QTable
+from astropy.units import UnitConversionError
+from . import __version__ as desiutilVersion
+from .log import get_logger, DEBUG
+
+
+log = None
+
+
+def csv_unit_column(header, comment=False):
```

Comment:
Thanks for supporting "unit" and "units", "comment" and "comments", and allowing this to be case insensitive.

minor maintainability: I found it confusing that a function called `csv_unit_column` used a boolean switch to search for the comments column name, and that the iterable of column names was called `header` (which implied to me a fits-style key/value header).  Consider updating this to something like
```python
def find_column_name(colnames, prefix='unit'):
    """find case insensitive column name starting with `prefix` from colnames"""
```

Reply:
OK, that's reasonable, but `prefix` has to be a `tuple` not a string, because if you look in more detail, sometimes the comments are called "Description".

Reply:
Done in [dab51a4](https://github.com/desihub/desiutil/pull/199/commits/dab51a474c40e19cc3dbbc9c2b8cfd0d4d4e199b).